v2/tools/generator/internal/config/type_configuration.go

```diff
@@ -106,6 +109,41 @@ func (tc *TypeConfiguration) VerifyExportAsConsumed() error {
 	return nil
 }
 
+// SetAzureGeneratedSecrets sets the list of Azure Generated secrets this type supports
+func (tc *TypeConfiguration) SetAzureGeneratedSecrets(secrets []string) *TypeConfiguration {
```

Comment:
I normally expect a `SetX()` method for a property to prune any existing values, but this method just adds them to whatever is there. 

Suggest either renaming to `AddAzureGeneratedSecrets()` or starting with an empty slice.

Reply:
Actually got rid of this method entirely since its functionality was subsumed into the new `configurableStringSlice`